### PR TITLE
Handle null list of files while providing segment size

### PR DESCRIPTION
**Pull request by antumbde**

### CODE REVIEW

File: pinot-server/src/main/java/com/linkedin/pinot/server/api/restlet/TableSizeResource.java [hunk 1]
```diff
@@ -86,20 +86,24 @@ public StringRepresentation getTableSize(
...
-        tableSizeInfo.segments.add(segmentSizeInfo);
+    try {
+      for (SegmentDataManager segmentDataManager : segmentDataManagers) {
+        IndexSegment segment = segmentDataManager.getSegment();
+        long segmentSizeBytes = segment.getDiskSizeBytes();
+        if (detailed) {
+          SegmentSizeInfo segmentSizeInfo = new SegmentSizeInfo(segment.getSegmentName(), segmentSizeBytes);
+          tableSizeInfo.segments.add(segmentSizeInfo);
+        }
+        tableSizeInfo.diskSizeInBytes += segmentSizeBytes;
```
Comment (jfim) [inline pinot-server/src/main/java/com/linkedin/pinot/server/api/restlet/TableSizeResource.java]: Should we exclude segments that have a size of `-1` here?

Comment (antumbde) [inline pinot-server/src/main/java/com/linkedin/pinot/server/api/restlet/TableSizeResource.java]: we shouldn't. -1 explicitly indicates error reading size. Segment missing in response will be ambiguous.

Controller explicitly handles -1.

**Comment by Jackie-Jiang** [comment]
In HTTP request, do we need to explicitly release the connection? If we don't release the connection, can the connection get reused for the next request? And after some time without request, the connection automatically closed?

**Comment by antumbde** [comment]
We are using connection manager so connections are re-used. Connection manager has a pool of 100 connections and it will recycle that for requests closing as required (or if stale). The getMethod.releaseConnection you see here is effectively releasing the connection back to the pool. It does not necessarily close the connection to the server. ConnectionManager is handling all that.

Normally, getMethod.releaseConnection is not required. Connection is automatically released after we read response from the server. However, in case of non-2xx code and some exception cases, we may not read complete response making getMethod to hold on to those connections and not release those to the pool. After some time, the pool starts to run out of connections slowing queries. This will eventually cause timeouts for getting a connection.

File: pinot-server/src/main/java/com/linkedin/pinot/server/api/restlet/TableSizeResource.java (hunk 1 above)
Comment (jfim) [inline pinot-server/src/main/java/com/linkedin/pinot/server/api/restlet/TableSizeResource.java]: So if we have a segment of 10000 bytes and a defunct segment, `diskSizeInBytes` should be 9999 bytes?

**Pull request merged**